boa3/builtin/__init__.py

```diff
@@ -123,24 +124,84 @@ def add_trusted_source(self, hash_or_address: str):
             return
 
         if hash_or_address == IMPORT_WILDCARD:
-            self._trusts.clear()
             self._trusts = [IMPORT_WILDCARD]
 
         # verifies if it's a valid contract hash
-        elif hash_or_address.startswith('0x'):
+        elif self.verify_is_valid_contract_hash(hash_or_address):
+            if hash_or_address not in self._trusts:
+                self._trusts.append(hash_or_address.lower())
+
+        # verifies if it's a valid public key
+        elif self.verify_is_valid_public_key(hash_or_address):
+            if hash_or_address not in self._trusts:
+                self._trusts.append(hash_or_address.lower())
+
+    def add_permission(self, *, contract: str = None, methods: Union[List[str], str] = None):
```

Comment:
the default value could be the wildcard directly
```suggestion
    def add_permission(self, *, contract: str = IMPORT_WILDCARD, methods: Union[List[str], str] = IMPORT_WILDCARD):
```